src/hypothesis/strategies.py

```diff
@@ -1575,4 +1578,204 @@ def deferred(definition):
+        # a valid utf-8 sequence when encoded and result in a different
+        # code point when decoded again. Can happen when two paths get
+        # concatenated. Shuffling makes it possible to generate such a case.
+        text_strategy = permutations(draw(unix_path_text)).map(u"".join)
+
+    if result_type is None:
+        return draw(one_of(bytes_strategy, text_strategy))
+    elif result_type is bytes:
+        return draw(bytes_strategy)
+    else:
+        return draw(text_strategy)
+
+
+def _str_to_path(s, result_type):
+    """Given an ASCII str, returns a path of the given type."""
+
+    assert isinstance(s, str)
+    if isinstance(s, bytes) and result_type is text_type:  # pragma: no cover
+        return s.decode('ascii')
+    elif isinstance(s, text_type) and result_type is bytes:
+        return s.encode('ascii')
+    return s
+
+
+@composite
+def _path_root(draw, result_type):
+    """Generates a root component for a path."""
+
+    # Based on https://en.wikipedia.org/wiki/Path_(computing)
+
+    def tp(s=''):
+        return _str_to_path(s, result_type)
+
+    if os.name != 'nt':
+        return tp(os.sep)
+    else:  # pragma: no cover
+        sep = sampled_from([os.sep, os.altsep or os.sep]).map(tp)
+        name = _filename(result_type)
+        char = name.map(lambda n: n[0:1])
+
+        relative = sep
+        # [drive_letter]:\
+        drive = builds(lambda *x: tp().join(x), char, just(tp(':')), sep)
+        # \\?\[drive_spec]:\
+        extended = builds(
+            lambda *x: tp().join(x), sep, sep, just(tp('?')), sep, drive)
+
+        network = one_of([
+            # \\[server]\[sharename]\
+            builds(lambda *x: tp().join(x), sep, sep, name, sep, name, sep),
+            # \\?\[server]\[sharename]\
+            builds(lambda *x: tp().join(x),
+                   sep, sep, just(tp('?')), sep, name, sep, name, sep),
+            # \\?\UNC\[server]\[sharename]\
+            builds(lambda *x: tp().join(x),
+                   sep, sep, just(tp('?')), sep, just(tp('UNC')), sep, name,
+                   sep, name, sep),
+            # \\.\[physical_device]\
+            builds(lambda *x: tp().join(x),
+                   sep, sep, just(tp('.')), sep, name, sep),
+        ])
+
+        return draw(one_of(relative, drive, extended, network))
+
+
+@defines_strategy
+@composite
+def fspaths(draw, allow_pathlike=None, allow_existing=False):
+    """A strategy which generates filesystem path values.
+
+    The generated values include everything which the builtin
+    :func:`python:open` function accepts i.e. which won't lead to
+    :exc:`ValueError` or :exc:`TypeError` being raised.
+
+    Note that the range of the returned values depends on the operating
+    system, the Python version, and the filesystem encoding as returned by
+    :func:`sys.getfilesystemencoding`.
+
+    :param allow_pathlike:
+        By default makes the strategy also include objects implementing the
```

Comment:
Rather than say "By default" I'd say "If None" as per the other possible argument values.